src/test/regress/sql/qp_dropped_cols.sql

```diff
@@ -8624,6 +8624,29 @@ explain select * from multi_part_mid_bitmap where date = '2017-02-05' and region
 -- end_ignore
 select * from multi_part_mid_bitmap where date = '2017-02-05' and region = 'usa';
 
+-- Exchange partitions with dropped columns and check subplan with attribute remapping into dynamicSeqscan
+CREATE TABLE ds_main ( a INT, b INT, c INT) PARTITION BY RANGE(c)( START(1) END (10) EVERY (2), DEFAULT PARTITION deflt);
+CREATE TABLE ds_part1 (a int, a1 int, b int, c int);
+CREATE TABLE non_part1 (c INT);
+CREATE TABLE non_part2 (e INT, f INT);
+
+SET optimizer_enforce_subplans TO ON;
+
+-- drop columns to get attribute remapping
+ALTER TABLE ds_part1 drop column a1;
+ALTER TABLE ds_main exchange partition for (1) with table ds_part1;
+
+INSERT INTO non_part1 SELECT i FROM generate_series(1, 100)i;
+INSERT INTO non_part2 SELECT i, i + 1 FROM generate_series(1, 100)i;
+INSERT INTO ds_main SELECT i, i + 1, i + 2 FROM generate_series (1, 1000)i;
+
+analyze ds_main;
+analyze non_part1;
+analyze non_part2;
+
+EXPLAIN (costs off) SELECT * FROM ds_main, non_part2 WHERE ds_main.c = non_part2.e AND non_part2.f < 10 AND a IN ( SELECT a FROM non_part1) order by a;
+SELECT * FROM ds_main, non_part2 WHERE ds_main.c = non_part2.e AND non_part2.f < 10 AND a IN ( SELECT a FROM non_part1) order by a;
```

Comment:
Predicate `non_part2.f < 10` is redundant here.
It's enough to keep 10 rows in `non_part2` table and 1 row in `non_part1`.